src/lib/hooks/useShareAsset.ts

```diff
@@ -0,0 +1,22 @@
+import { useCallback } from 'react';
+
+import { NFT_ASSET_SHARE_KEYS, shareDomainAsset } from '../helpers';
+
+interface useShareAssetReturn {
+	shareAsset: (key: NFT_ASSET_SHARE_KEYS) => Promise<void>;
+}
+
+export const useShareAsset = (zna: string): useShareAssetReturn => {
+	const shareAsset = useCallback(
+		async (key: NFT_ASSET_SHARE_KEYS) => {
+			if (zna) {
+				shareDomainAsset(zna, key);
+			}
+		},
+		[zna],
+	);
+
+	return {
+		shareAsset,
+	};
+};
```

Comment:
Similar w `useDownloadAsset` - could be a utility function as it doesn't tie into any state/lifecycle.